File: examples/canonical-faker/java/src/test/java/io/agentcontextbase/faker/SmokeTest.java
package io.agentcontextbase.faker;

import io.agentcontextbase.faker.domain.TenantCoreEntities;
import io.agentcontextbase.faker.pipeline.TenantCorePipeline;
import io.agentcontextbase.faker.profiles.Profile;
import io.agentcontextbase.faker.validate.ValidationReport;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class SmokeTest {
    @Test
    void smokeProducesCorrectOrgCount() {
        TenantCoreEntities.TenantCoreDataset dataset = TenantCorePipeline.generate(Profile.SMOKE);
        assertEquals(3, dataset.organizations().size());
    }

    @Test
    void smokePassesValidation() {
        TenantCoreEntities.TenantCoreDataset dataset = TenantCorePipeline.generate(Profile.SMOKE);
        ValidationReport report = TenantCorePipeline.validate(dataset);
        assertTrue(report.ok(), "Violations: " + report.violations());
    }

    @Test
    void smokeIsReproducible() {
        TenantCoreEntities.TenantCoreDataset left = TenantCorePipeline.generate(Profile.SMOKE);
        TenantCoreEntities.TenantCoreDataset right = TenantCorePipeline.generate(Profile.SMOKE);
        assertEquals(left.organizations(), right.organizations());
    }
}

File: examples/canonical-faker/kotlin/src/test/kotlin/io/agentcontextbase/faker/SmokeTest.kt
package io.agentcontextbase.faker

import io.agentcontextbase.faker.pipeline.KotlinFakerPipeline
import io.agentcontextbase.faker.profiles.Profile
import io.agentcontextbase.faker.validate.validate
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class SmokeTest {
    @Test
    fun smokeProducesCorrectOrgCount() {
        val dataset = KotlinFakerPipeline.generate(Profile.SMOKE)
        assertEquals(3, dataset.organizations.size)
    }

    @Test
    fun smokePassesValidation() {
        val dataset = KotlinFakerPipeline.generate(Profile.SMOKE)
        val report = validate(dataset)
        assertTrue(report.ok, "Violations: ${report.violations}")
    }

    @Test
    fun smokeIsReproducible() {
        val left = KotlinFakerPipeline.generate(Profile.SMOKE)
        val right = KotlinFakerPipeline.generate(Profile.SMOKE)
        assertEquals(left.organizations, right.organizations)
    }
}
